src/tests/ftest/util/daos_io_conf.py

```diff
@@ -0,0 +1,112 @@
+"""
+import os
+from avocado.utils import process
+
+from apricot import TestWithServers
+from command_utils import ExecutableCommand, CommandFailure, FormattedParameter
+from command_utils import BasicParameter
+from test_utils import TestPool
+
+class IoConfGen(ExecutableCommand):
+    """Defines an object for the daos_gen_io_conf and daos_run_io_conf commands.
+
+    :avocado: recursive
+    """
+    def __init__(self, path="", env=None):
+        """Create a ExecutableCommand object.
+
+        Uses Avocado's utils.process module to run a command str provided.
+
+        Args:
+            command (str): string of the command to be executed.
+            path (str, optional): path to location of command binary file.
+                Defaults to ""
+        """
+        super(IoConfGen, self).__init__("/run/run_gen_io_conf/*",
+                                        "daos_gen_io_conf")
+        self.verbose = True
+        self.env = env
+        self.ranks = FormattedParameter("-g {}")
+        self.targets = FormattedParameter("-t {}")
+        self.obj_num = FormattedParameter("-o {}")
+        self.akeys = FormattedParameter("-a {}")
+        self.dkeys = FormattedParameter("-d {}")
+        self.re_size = FormattedParameter("-s {}")
+        self.obj_class = FormattedParameter("-O {}")
+        self.filename = BasicParameter(None, "testfile")
+
+    def run_conf(self):
+        """Run the daos_run_io_conf command as a foreground process.
+
+        Raises:
+            CommandFailure: if there is an error running the command
+
+        """
+        command = " ".join([os.path.join(self._path, "daos_run_io_conf"),
+                            self.filename.value])
+        kwargs = {
+            "cmd": command,
+            "timeout": self.timeout,
+            "verbose": self.verbose,
+            "allow_output_check": "combined",
+            "shell": True,
+            "env": self.env,
+            "sudo": self.sudo,
+        }
+        try:
+            # Block until the command is complete or times out
+            return process.run(**kwargs)
+
+        except process.CmdError as error:
+            # Command failed or possibly timed out
+            msg = "Error occurred running '{}': {}".format(command, error)
+            self.log.error(msg)
+            raise CommandFailure(msg)
+
+class IoConfTestBase(TestWithServers):
+    """Base rebuild test class.
+
+    :avocado: recursive
+    """
+
+    def __init__(self, *args, **kwargs):
+        """Initialize a IoConfTestBase object."""
+        super(IoConfTestBase, self).__init__(*args, **kwargs)
+
+    def setup_test_pool(self):
+        """Define a TestPool object."""
+        self.pool = TestPool(self.context, self.log)
+        self.pool.get_params(self)
+
+    def execute_io_conf_run_test(self):
+        """
+        Execute the rebuild test steps.
+        """
+        self.setup_test_pool()
+        io_conf = IoConfGen()
```

Comment:
Should we be explicitly passing in a path here? Without it are we not relying on `daos_[gen|run]_io_conf` to be in the PATH?

For example:
```
io_conf = IoConfGen(self.bin)
```
or (if #1008 hasn't landed yet):
```
io_conf = IoConfGen(os.path.join(self.prefix, "bin"))
```